### ISSUE TITLE
Setting negative margin to 0 in contrastive loss with cosine similarity is wrong?

### ISSUE THREAD

**Issue by dvirginz** [KevinMusgrave/pytorch-metric-learning#246]
Thank you for this great repo!
In the constructive loss documentation, you advise setting the positive margin to 1 and the negative margin 0 when using cosine similarity as the metric. [Constractive](https://kevinmusgrave.github.io/pytorch-metric-learning/losses/#contrastiveloss)

Unfortunately, I'm not sure I understand this, as for any cosine similarity bigger than 0 the negative loss is zero (0 - positive value)
and even worst, for negative cosine similarities (good for negative samples) the loss is positive(!) bad.
Am I missing something?

**Comment by KevinMusgrave** [comment]
When using a similarity metric, the equation changes to:

```
loss = [m_pos - s_p] + [s_n - m_neg]
```

So for a negative margin of 0, the loss for the negative component will be ```[s_n]```

I'll update the documentation to explain this.

**Comment by dvirginz** [comment]
I see that now in the code (cosine similarity has the `is_inverted` flag on by default)
